.github/workflows/tools_tests.yaml

```
@@ -39,6 +39,25 @@ jobs:
                         source selfdrive/test/setup_vsound.sh && \
                         CI=1 pytest tools/sim/tests/test_metadrive_bridge.py"
 
+  test_python311:
+    name: test python3.11 support
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
```

this needs to be super fast!
```suggestion
    timeout-minutes: 5
```